multidict/_multidict.c

```diff
@@ -78,7 +78,7 @@ _multidict_extend(MultiDictObject *self, PyObject *arg,
     }
 
     if (arg != NULL) {
-        if (MultiDict_CheckExact(arg) || CIMultiDict_CheckExact(arg)) {
+        if (CIMultiDict_CheckExact(arg) || (!CIMultiDict_CheckExact(self) && MultiDict_CheckExact(arg))) {
```

Comment:
This isn't efficient because it ends up falling through to `PyMapping_Items`

Reply:
Please no, I prefer the proper fix.

Reply:
Thanks for finishing it up 👍